test/MarketMaker.js

```diff
@@ -162,6 +215,70 @@ contract('marketMaker', function(accounts) {
     await help.waitToBlock(startBlock+blocksPerPeriod*numberOfMonth, accounts);
   };
 
+  it("Should return the correct sellPrice on every period with 873750 tokens total supply and 3285 ETH in MM", async function() {
+    token = await simulateCrowdsale(100, [873750,0,0,0,0], accounts);
+
+    const startBlock = web3.eth.blockNumber+10;
+    const blocksPerPeriod = 5;
+    const initialPriceSpread = 0.05;
+    const priceFactor = 100000;
+    const initialPriceSpreadFactor = priceFactor*(1+initialPriceSpread);
+    const initialMMEther = 30285;
+    const MMInitialBalance = web3.toWei(initialMMEther, 'ether');
+    const tokenTotalSupply = help.lif2LifWei(873750);
+
+    const initialSellPrice = parseFloat((MMInitialBalance / tokenTotalSupply)*(1+initialPriceSpread)).toFixed(5);
+    const initialBuyPrice = parseFloat((MMInitialBalance / tokenTotalSupply)).toFixed(5);
+
+    mm = await LifMarketMaker.new(
+      token.address, startBlock, blocksPerPeriod, 24,
+      accounts[1], initialPriceSpreadFactor,
+      {value: web3.toWei(initialMMEther, 'ether'), from: accounts[0]}
+    );
+
+    await mm.calculateDistributionPeriods({from: accounts[4]});
+    await mm.calculateSellPricePeriods({from: accounts[3]});
+
+    help.debug('Total Token Supply:', help.lifWei2Lif(parseFloat( await token.totalSupply())));
+    help.debug('MM balance:', parseFloat( web3.eth.getBalance(mm.address) ).toFixed(0));
+    help.debug('Start block', parseInt( await mm.startBlock.call() ));
+    help.debug('Blocks per period', parseInt( await mm.blocksPerPeriod.call() ));
+    help.debug('Foundation address', await mm.foundationAddr.call() );
+    help.debug('Initial Wei', parseFloat( await mm.initialWei.call() ).toFixed(0));
+
+    assert.equal(initialBuyPrice, parseFloat(await mm.initialBuyPrice()/priceFactor).toFixed(5) );
+    assert.equal(initialSellPrice, parseFloat(await mm.initialSellPrice()/priceFactor).toFixed(5) );
+
+    let accumIncrementPrice = [
+      0, 1000, 2010, 3030, 4060, 5101,
+      6152, 7213, 8285, 9368, 10462,
+      11566, 12682, 13809, 14947, 16096,
+      17257, 18430, 19614, 20810, 22019,
+      23239, 24471, 25716, 26973, 28243,
+      29525, 30820, 32129, 33450, 34784,
+      36132, 37494, 38869, 40257, 41660,
+      43076, 44507, 45952, 47412, 48886,
+      50375, 51878, 53397, 54931, 56481,
+      58045, 59626
+    ];
+
+    await help.waitToBlock(startBlock+1);
+    for (var i = 0; i < 24; i++) {
+
+      help.debug('Sell price on period', i,
+        parseFloat(await mm.getSellPrice()/priceFactor)
+      );
+
+      assert.approximately(
+        parseFloat(initialSellPrice*(1.01**i))*priceFactor/priceFactor,
```

Comment:
What's the reason behind `*priceFactor/priceFactor`? Can't we remove that term?

Perhaps we should keep all the numbers multiplied by priceFactor, so we don't have to divide what we get back from the contract by `priceFactor` like is done in the next line

Also, I'd still like to see some assertions (at least 1, maybe 3 would be nice) with concrete numbers (say what's the price at period 1, what's the price at period 14, what's the price at period 23)